api/institutions/parsers.py

```diff
@@ -0,0 +1,29 @@
+import json
+import httplib
+
+import jwe
+import jwt
+from rest_framework.parsers import BaseParser
+
+from api.base import settings
+from framework.exceptions import HTTPError
+
+class InstitutionAuthParser(BaseParser):
```

Comment:
I'm not sure that it makes sense to use a `Parser` to do this work. I think it makes sense to implement a drf `Authentication` class instead. See http://www.django-rest-framework.org/api-guide/authentication/